src/MagicStringOffset.js

```diff
@@ -0,0 +1,90 @@
+import MagicString from './MagicString.js';
+
+export default class MagicStringOffset {
```

Comment:
Should we use extends instead? This way it would still be treated like a normal MagicString instance, and `instance of MagicString` would still work

Reply:
Ah, I see you did the magics here 🤣